homeassistant/components/onvif/binary_sensor.py

```diff
@@ -45,43 +45,17 @@ class ONVIFBinarySensor(ONVIFBaseEntity, BinarySensorEntity):
 
     def __init__(self, uid, device):
         """Initialize the ONVIF binary sensor."""
-        ONVIFBaseEntity.__init__(self, device)
-        BinarySensorEntity.__init__(self)
-
-        self.uid = uid
-
-    @property
-    def is_on(self) -> bool:
-        """Return true if event is active."""
-        return self.device.events.get_uid(self.uid).value
-
-    @property
-    def name(self) -> str:
-        """Return the name of the event."""
-        return self.device.events.get_uid(self.uid).name
-
-    @property
-    def device_class(self) -> str | None:
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return self.device.events.get_uid(self.uid).device_class
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return self.uid
-
-    @property
-    def entity_registry_enabled_default(self) -> bool:
-        """Return if the entity should be enabled when first added to the entity registry."""
-        return self.device.events.get_uid(self.uid).entity_enabled
-
-    @property
-    def should_poll(self) -> bool:
-        """Return True if entity has to be polled for state.
-
-        False if entity pushes its state to HA.
-        """
-        return False
+        self._attr_device_class = device.events.get_uid(uid).device_class
```

Comment:
Why don't you once call `device.events.get_uid(uid)` and simplify the rest of the code ?